src/main/java/htsjdk/variant/variantcontext/writer/VCFWriter.java

```diff
@@ -228,4 +228,10 @@ public void add(final VariantContext context) {
             throw new RuntimeIOException("Unable to write the VCF object to " + getStreamName(), e);
         }
     }
+
+    @Override
+    public void setVcfHeader(VCFHeader header) {
+        this.mHeader = doNotWriteGenotypes ? new VCFHeader(header.getMetaDataInSortedOrder()) : header;
+        this.vcfEncoder = new VCFEncoder(this.mHeader, this.allowMissingFieldsInHeader, this.writeFullFormatField);
```

Comment:
@cmnbroad @droazen @yfarjoun Added the same checks for BCF2Writer as well. I think with this commit I've addressed all the comments so far. Waiting for further feedback.